compositor_pipeline/src/pipeline/output.rs

```diff
@@ -124,8 +124,31 @@ impl OutputOptionsExt<Option<Port>> for OutputOptions {
                 Ok((Output::Rtp { sender, encoder }, Some(port)))
             }
             OutputProtocolOptions::Mp4(mp4_opt) => {
-                let writer = Mp4FileWriter::new(output_id.clone(), mp4_opt.clone(), packets, ctx)
-                    .map_err(|e| RegisterOutputError::OutputError(output_id.clone(), e))?;
+                let encoder_config = Mp4EncoderConfig {
+                    video: encoder.video.as_ref().map(|_| ()),
+                    audio: encoder
+                        .audio
+                        .as_ref()
+                        .map(|encoder| match encoder {
+                            AudioEncoder::Opus(_opus_encoder) => {
+                                Err(RegisterOutputError::UnknownError(
```

Comment:
I added an unknown error because this would be a bug in our code if this triggered.